src/SpeedyTransforms/scratch_memory.jl

```diff
@@ -0,0 +1,50 @@
+"""
+ScratchMemory holds scratch memory for the `SpectralTransform` that's used both by the Fourier and Legendre transform. Fields are
+$(TYPEDFIELDS)"""
+struct ScratchMemory{
+    NF,
+    ArrayComplexType,           # <: ArrayType{Complex{NF}, 3},
+}
+    # SCRATCH MEMORY FOR FOURIER NOT YET LEGENDRE TRANSFORMED AND VICE VERSA
+    # state is undetermined, only read after writing to it
+    north::ArrayComplexType
+    south::ArrayComplexType
+end 
+
+function ScratchMemory(
+    ::Type{NF},                     
+    ArrayType::Type{<:AbstractArray}, 
+    nfreq_max::Integer, 
+    nlayers::Integer, 
+    nlat_half::Integer) where NF
+
+    # SCRATCH MEMORY FOR FOURIER NOT YET LEGENDRE TRANSFORMED AND VICE VERSA
+    scratch_memory_north = zeros(Complex{NF}, nfreq_max, nlayers, nlat_half)
+    scratch_memory_south = zeros(Complex{NF}, nfreq_max, nlayers, nlat_half)
+
+    return ScratchMemory{
+        NF,
+        ArrayType{Complex{NF}, 3}
+    }(scratch_memory_north, scratch_memory_south)
+end 
+
+"""$(TYPEDSIGNATURES)
+Generator function for a `ScratchMemory` that holds the scratch memory for SpeedyTransforms.
+"""
+function ScratchMemory(
```

Comment:
should there be a constructor method to create a 0-element `ScratchMemory` which could be in `SpectralTransform` if the scratch space in there is not needed? E.g. in the normal case of running a simulation the scratch memory can just be in the diagnostic variables and so the scratch memory in the spectral transform shouldn't be allocated?

Reply:
This is now resolved through reference to the scratch memory